### PR TITLE
feat: migrate helm charts repo from odigos-charts to here

### PR DESCRIPTION
**Pull request by rauno56**
After merging this we will be managing helm charts in this repo fully. The old index will then stay up and working but will be stale.

This has a benefit of being able to update helm and code in the same PR, simplifying the whole release process and having e2e tests run on the latest unreleased helm version.

This PR will also change the chart versioning(currently `0.3.x`) to match the application version(currently `1.0.x`) on every release.

This includes [2bd8d7b](https://github.com/odigos-io/odigos-charts/commit/2bd8d7b9f8ec38a5acad52bd0b1ad17904f5a850) of odigos-charts which is the last commit at the moment of creation of this PR.

Commits:

- **feat: migrate helm chart over as is**
- **feat: default image tags to Chart.AppVersion**
- **feat: point references to external chart repository to this repo**

### CODE REVIEW

**Review by blumamir — APPROVED** [review]
LGTM 👍 

@edeNFed  do you want to have a look as well?

File: tests/e2e/helm-chart/chainsaw-test.yaml [hunk 1]
```diff
@@ -25,14 +25,12 @@ spec:
     try:
     - script:
         content: |
-          git clone https://github.com/odigos-io/odigos-charts.git /tmp/odigos-charts
           # Retry to avoid flakiness (due to CRD race conditions in helm). Timeout after 60s.
-          while ! helm upgrade --install odigos /tmp/odigos-charts/charts/odigos --create-namespace --namespace odigos-test-ns --set image.tag=e2e-test; do
+          while ! helm upgrade --install odigos $GITHUB_WORKSPACE/helm/odigos --create-namespace --namespace odigos-test-ns --set image.tag=e2e-test; do
```
Comment (edeNFed) [inline tests/e2e/helm-chart/chainsaw-test.yaml:29]: nit: I used `/tmp` to easily be able to run chainsaw tests locally outside Github Actions

**Review by edeNFed — APPROVED** [review]

File: tests/e2e/helm-chart/chainsaw-test.yaml (hunk 1 above)
Comment (rauno56) [reply tests/e2e/helm-chart/chainsaw-test.yaml:29]: I understand that.

We need to somehow find the project root. Chainsaw just changes the current working directory and, outside of GHA, I don't even know how to retrieve it.

Do you know a way to reference project root in chainsaw tests that is not GHA-specific?

Comment (rauno56) [reply tests/e2e/helm-chart/chainsaw-test.yaml]: Changed in https://github.com/odigos-io/odigos/pull/1429/commits/c44935bec2823654ebc87fd01d11c2b6fcac5c4e

**Pull request merged**